client/my-sites/stats/stats-post-detail/index.jsx

```diff
@@ -37,16 +38,14 @@ export default React.createClass( {
 	render() {
 		let title;
 
-		if ( this.props.postViewsList.response.post && this.props.postViewsList.response.post.post_title ) {
-			title = this.translate( 'Stats for %(posttitle)s', {
-				comment: 'Title of the individual post stats page.',
-				args: {
-					posttitle: this.props.postViewsList.response.post.post_title
-				}
-			} );
-		}
+		const post = this.props.postViewsList.response.post;
+		const postOnRecord = post && post.post_title !== null;
 
-		if ( this.props.postViewsList.isError() ) {
+		if ( postOnRecord ) {
+			if ( typeof post.post_title === 'string' && post.post_title.length ) {
```

Comment:
So having this double nested if case right here could still yield a scenario where `title` is null.  I think moving the length check to line 44 would be sufficient since postOnRecord is already doing a null check.